src/dragonball/src/device_manager/console_manager.rs

```diff
@@ -0,0 +1,427 @@
+        self.sock_path = Some(sock_path);
+
+        Ok(())
+    }
+
+    /// Reset the host side terminal to canonical mode.
+    pub fn reset_console(&self) -> Result<()> {
+        if let Some(stdin_handle) = self.stdin_handle.as_ref() {
+            stdin_handle
+                .lock()
+                .set_canon_mode()
+                .map_err(|e| DeviceMgrError::ConsoleManager(ConsoleManagerError::StdinHandle(e)))?;
+        }
+
+        Ok(())
+    }
+
+    fn bind_domain_socket(serial_path: &str) -> std::result::Result<UnixListener, std::io::Error> {
+        let path = Path::new(serial_path);
+        if path.is_file() {
+            let _ = std::fs::remove_file(serial_path);
+        }
+
+        UnixListener::bind(path)
+    }
+}
+
+struct ConsoleEpollHandler {
+    device: Arc<Mutex<SerialDevice>>,
+    stdin_handle: Option<std::io::Stdin>,
+    sock_listener: Option<UnixListener>,
+    sock_conn: Option<UnixStream>,
+    logger: slog::Logger,
+}
+
+impl ConsoleEpollHandler {
+    fn new(
+        device: Arc<Mutex<SerialDevice>>,
+        stdin_handle: Option<std::io::Stdin>,
+        sock_listener: Option<UnixListener>,
+        logger: &slog::Logger,
+    ) -> Self {
+        ConsoleEpollHandler {
+            device,
+            stdin_handle,
+            sock_listener,
+            sock_conn: None,
+            logger: logger.new(slog::o!("subsystem" => "console_manager")),
+        }
+    }
+
+    fn uds_listener_accept(&mut self, ops: &mut EventOps) -> std::io::Result<()> {
+        if self.sock_conn.is_some() {
+            slog::warn!(self.logger,
+                "UDS for serial port 1 already exists, reject the new connection";
+                "subsystem" => "console_mgr",
+            );
+            // Do not expected poisoned lock.
+            let _ = self.sock_listener.as_mut().unwrap().accept();
+        } else {
+            // Safe to unwrap() because self.sock_conn is Some().
+            let (conn_sock, _) = self.sock_listener.as_ref().unwrap().accept()?;
+            let events = Events::with_data(&conn_sock, EPOLL_EVENT_SERIAL_DATA, EventSet::IN);
+            if let Err(e) = ops.add(events) {
+                slog::error!(self.logger,
+                    "failed to register epoll event for serial, {:?}", e;
+                    "subsystem" => "console_mgr",
+                );
+                return Err(std::io::Error::last_os_error());
+            }
+
+            let conn_sock_copy = conn_sock.try_clone()?;
+            // Do not expected poisoned lock.
+            self.device
+                .lock()
+                .unwrap()
+                .set_output_stream(Some(Box::new(conn_sock_copy)));
+
+            self.sock_conn = Some(conn_sock);
+        }
+
+        Ok(())
+    }
+
+    fn uds_read_in(&mut self, ops: &mut EventOps) -> std::io::Result<()> {
+        let mut should_drop = true;
+
+        if let Some(conn_sock) = self.sock_conn.as_mut() {
+            let mut out = [0u8; 64];
```

Comment:
Could you make this a `const` with a comment explaining why 64 was chosen?

Reply:
Changed.